torchtitan/checkpoint.py

```diff
@@ -67,6 +70,33 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         set_optimizer_state_dict(self.model, self.optim, optim_state_dict=state_dict)
 
 
+class DataLoaderWrapper(Stateful):
+    def __init__(self, dataloader: DataLoader) -> None:
+        self.dataloader = dataloader
+        # Use global rank for now even though dataloader state could be same across dp groups
+        self.rank_id = str(
+            dist.get_rank() if (dist.is_available() and dist.is_initialized()) else 0
+        )
+
+    def state_dict(self) -> Dict[str, Any]:
+        if isinstance(self.dataloader, StatefulDataLoader):
+            return {self.rank_id: pickle.dumps(self.dataloader.state_dict())}
+        return {}
+
+    def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
+        if isinstance(self.dataloader, StatefulDataLoader):
+            # State is empty
+            if not state_dict:
+                return
+
+            if self.rank_id not in state_dict:
+                logger.warning(f"DataLoader state is empty for rank {self.rank_id}. ")
+                return
+
+            # Load state for the current rank
+            self.dataloader.load_state_dict(pickle.loads(state_dict[self.rank_id]))
```

Comment:
wow is this made possible by recent change from StatefulDataloader? What nightly version can we use to run it?

Reply:
Sorry what is made possible? This is just deserializing the byte string of dataloader state back to a state_dict and passing it in.